libraries/adopt_openjdk_helpers.rb

```diff
@@ -22,14 +22,19 @@ def sub_dir(url)
           result
 
         else # custom url
-          # assuming URL like url.of/custom/location/OpenJDK$VER-jdk_x64_linux_$VARIANT_$HASH.tar.gz
-          # get file basename, remove extention, get hash
-          hash = url.split('/')[-1].split('.')[0].split('_')[-1]
+          # get file basename without extension
+          basename = url.split('/')[-1].gsub('.tar.gz', '')
 
-          raise("Failed to parse #{f} for directory name!") if hash.empty?
+          # Get version number from start of filename
+          case (basename.scan /\d+/)[0]
+          when '8'
+            ver = basename.split('_')[-1]
+            "jdk#{ver[0..4]}-#{ver[-3..-1]}"
+          else
+            ver = basename.split('_')
+            "jdk-#{ver[-2]}+#{ver[-1]}"
```

Comment:
I was just playing around in irb and this line should support both hotspot and openj9 if rewritten as:

```ruby
"jdk-#{ver[4]}+#{ver[5]}"
```